src/js/app/modules/dashboard/DashboardController.js

```diff
@@ -1,11 +1,67 @@
-app.controller('DashboardController', function($scope, $location, DashboardService, ChartService){
+app.controller('DashboardController', function($scope, $location, $stateParams, DashboardService, ChartService){
 
   var mainChart, relationsChart;
+  $scope.selected = $stateParams.hostname;
 
-  // update data when selected host has been changed
-  $scope.$watch('selected', function(selectedHost, oldValue){
-    if (selectedHost) {
-      ChartService.getData(selectedHost.name).then(function(data){
+  if ($scope.selected){
+    loadData();
+  }
+  else{
+    DashboardService.getHosts().then(function(obj){
+      $scope.hosts = obj.data;
+      if(obj.data && angular.isArray(obj.data)){
+        $scope.selected = obj.data[0].name;
+        $location.url("/dashboard/" + $scope.selected); 
+      }
+    });
+  }
+
+  /**
+   * Mark a host as selected
+   */
+  $scope.select = function(host){
+    if (host){
+      console.log('Select host ' + host.name);
+      $scope.selected = host.name;
+    }
+  };
+
+  // TODO: Redirect to a host when none is selected #29
+  // there is no selected host
+  // $location.url("/dashboard/" + $scope.hosts[0]);
+
+  /**
+   * Calculate the totals for errors / warnings and info based on the data provided
+   * The totals will be set on the scope
+   */
+  function calculateTotals(data){
+    if (data && angular.isArray(data.columns)){
+      copy = angular.copy(data.columns); // copy array, so we can change it 
+      copy.forEach(function(item,i){
+        var type = item.shift(),
+        total = item.reduce(function(prev, current) {
+          return prev+current;
+        });
+        
+        switch(type){
+          case 'errors':
+            $scope.errorTotals = total;
+            break;
+          case 'warnings':
+            $scope.warningTotal = total;
+            break;
+          case 'info':
+            $scope.infoTotal = total;
+            break;
+          default:
+            console.error("No such data: " + info);
```

Comment:
Mixed double and single quotes.<br>'console' is not defined.<br>'info' is not defined.